### PR TITLE
Zelda mobile nav

### PR DESCRIPTION
**Pull request by novaugust**
Rebased version of #5337, with an additional commit to unify nav properties and rename actions. If things look good I'll squash my commits down

### CODE REVIEW

File: core/client/app/styles/components/settings-menu.css [hunk 1]
```diff
@@ -153,14 +153,20 @@ body.settings-menu-expanded .settings-menu-container {
...
-    z-index: 600;
+    z-index: 1200;
     transition: transform 0.4s cubic-bezier(0.1, 0.7, 0.1, 1);
-    transform: translate3d(-350px, 0px, 0px);
     /* Not off the screen, to give a parallax effect */
 }
+.settings-menu-expanded .eater-of-clicks-devourer-of-worlds-mother-of-dragons {
+    transform: translate3d(-350px, 0px, 0px);
+}
+.mobile-menu-expanded .eater-of-clicks-devourer-of-worlds-mother-of-dragons {
```
Comment (novaugust) [inline core/client/app/styles/components/settings-menu.css]: Note to self: rename this while rebasing...

Comment (ErisDS) [inline core/client/app/styles/components/settings-menu.css]: Could always shorten it down to '.devouring-mother' - reference to Defiance is less popular but still geeky :grin:

**Comment by jaswilli** [comment]
Giving this a[nother] run through and there's some weirdness with the nav menu and the post settings menu.  If you have the auto open/close behavior turned on and you open the PSM, the autonav menu won't open anymore.  You also can't close the PSM by clicking outside of it anymore.  There's an exception in the console about the `resetView` action not being handled.

Same thing in the Tags page.

**Comment by jaswilli** [comment]
And now I'm testing it on an iPhone 6 and I'm not seeing any menu at all?

**Comment by JohnONolan** [comment]
The whole thing is quite-a-lot broken for me. I think too much overlap of different people doing different things? In particular doesn't take into account viewport size changes and requires multiple refreshes to get the desired behaviour to "kick in" again.

**Comment by JohnONolan** [comment]
Also, the toggle detecting when the cursor enters `.gh-main` to close `.gh-nav` is inadvertently also closing all dropdown menus (which should not happen)

**Comment by novaugust** [comment]
John, can you give me a repro / gif? I'm not seeing the issues you are

**Comment by novaugust** [comment]
@jaswilli There's no way to make the autonav work with the PSM, because the PSM functions via the content-cover div which covers up gh-main and gh-nav. We can z-index gh-nav over the cover, allowing us to open the menu, but then mousing out won't reclose the menu as you're mousing into content-cover rather than gh-main.
Fixed the resetView bit, that was a last action name-change that I failed to find-replace-all of ;)

**Comment by jaswilli** [comment]
> make the autonav work with the PSM

Has something changed that makes the way I had it working not viable any longer?

**Comment by jaswilli** [comment]
> inadvertently also closing all dropdown menus

If we're talking about the same thing that was me.  Otherwise the nav menu can close with the user profile dropdown open, which is completely broken.

I think we can target dropdowns individually for closing so that should be pretty easy to correct so that dropdown gets closed but any others remain open.

**Comment by novaugust** [comment]
I mean, the scenario I just described... ah. It could be that the
`content-cover` use to be inside of gh-main instead of at the same scope.
Derp derp derp, I reckon that's fixable xD

**Comment by JohnONolan** [comment]
> If we're talking about the same thing that was me. Otherwise the nav menu can close with the user profile dropdown open, which is completely broken.

The dropdown is closing even when the nav menu isnt:

mobilemenu

**Comment by jaswilli** [comment]
Yeah... needs some fixing up :smile:

**Comment by jaswilli** [comment]
Still broken but merging so we have one fewer moving part.

**Pull request merged**